src/FastExpressionCompiler.LightExpression/Expression.cs

```diff
@@ -506,9 +506,18 @@ public abstract class BinaryExpression : Expression
         protected BinaryExpression(ExpressionType nodeType, Expression left, Expression right, Type type)
         {
             NodeType = nodeType;
-            Type = type;
+           
             Left = left;
             Right = right;
+
+            if (nodeType == ExpressionType.Equal || nodeType == ExpressionType.NotEqual || 
```

Comment:
Probably better to move this check to specific BE non-abstract implementor. But let's that be, will optimize later.